drivers/sensor/mpu9250/mpu9250_trigger.c

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2021, Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <drivers/sensor.h>
+#include <drivers/i2c.h>
+#include <drivers/gpio.h>
+#include <sys/util.h>
+#include <kernel.h>
+#include <logging/log.h>
+
+#include "mpu9250.h"
+
+LOG_MODULE_DECLARE(MPU9250, CONFIG_SENSOR_LOG_LEVEL);
+
+
+#define MPU9250_REG_INT_EN	0x38
+#define MPU9250_DRDY_EN		BIT(0)
+
+
+int mpu9250_trigger_set(const struct device *dev,
+			const struct sensor_trigger *trig,
+			sensor_trigger_handler_t handler)
+{
+	struct mpu9250_data *drv_data = dev->data;
+	const struct mpu9250_config *cfg = dev->config;
+	int ret;
+
+	if (trig->type != SENSOR_TRIG_DATA_READY) {
+		return -ENOTSUP;
+	}
+
+	ret = gpio_pin_interrupt_configure_dt(&cfg->int_pin, GPIO_INT_DISABLE);
+	if (ret < 0) {
+		LOG_ERR("Failed to disable gpio interrupt.");
+		return ret;
+	}
+
+	drv_data->data_ready_handler = handler;
+	if (handler == NULL) {
+		return 0;
+	}
+
+	drv_data->data_ready_trigger = *trig;
+
+	ret = gpio_pin_interrupt_configure_dt(&cfg->int_pin,
+					      GPIO_INT_EDGE_TO_ACTIVE);
+	if (ret < 0) {
+		LOG_ERR("Failed to enable gpio interrupt.");
+		return ret;
+	}
+
+	return 0;
+}
+
+static void mpu9250_gpio_callback(const struct device *dev,
+				  struct gpio_callback *cb, uint32_t pins)
+{
+	struct mpu9250_data *drv_data =
+		CONTAINER_OF(cb, struct mpu9250_data, gpio_cb);
+	const struct mpu9250_config *cfg = drv_data->dev->config;
+	int ret;
+
+	ARG_UNUSED(pins);
+
+	ret = gpio_pin_interrupt_configure_dt(&cfg->int_pin, GPIO_INT_DISABLE);
+	if (ret) {
+		LOG_ERR("Disabling gpio interrupt failed with err: %d", ret);
+	}
+
+#if defined(CONFIG_MPU9250_TRIGGER_OWN_THREAD)
+	k_sem_give(&drv_data->gpio_sem);
+#elif defined(CONFIG_MPU9250_TRIGGER_GLOBAL_THREAD)
+	k_work_submit(&drv_data->work);
+#endif
+}
+
+static void mpu9250_thread_cb(const struct device *dev)
+{
+	struct mpu9250_data *drv_data = dev->data;
+	const struct mpu9250_config *cfg = dev->config;
+	int ret;
+
+	if (drv_data->data_ready_handler != NULL) {
+		drv_data->data_ready_handler(dev,
+					     &drv_data->data_ready_trigger);
+	}
+
+	ret = gpio_pin_interrupt_configure_dt(&cfg->int_pin,
+					      GPIO_INT_EDGE_TO_ACTIVE);
+	if (ret) {
```

Comment:
please be consistent with other parts: `if (ret < 0)`

Reply:
Changed